docs/federation/federation-directives-index.mdx

```diff
@@ -0,0 +1,31 @@
+
+| Directive                        | Explanation                                                                                                                                        |
+| -------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------- |
+| [`@authenticated`](/federation/directives/authenticated)                 | Marks a field or type as requiring authentication. Only requests with valid credentials can access it.                                             |
+| `@composeDirective` (planned)    | Tells the composition process to preserve a specific custom type system directive in the supergraph.                                               |
+| [`@edfs__kafkaPublish`](/federation/event-driven-federation/kafka)            | Marks a mutation field that publishes an event to a Kafka topic. Requires a topic and provider ID.                                                 |
+| [`@edfs__kafkaSubscribe`](/federation/event-driven-federation/kafka)          | Declares a subscription field that listens to one or more Kafka topics. Used in EDG Subscription fields.                                           |
+| [`@edfs__natsPublish`](/federation/event-driven-federation/nats)             | Marks a mutation field that publishes an event to a NATS subject. Requires a subject and provider ID.                                              |
+| [`@edfs__natsRequest`](/federation/event-driven-federation/nats)             | Declares a query field that requests an event over NATS and expects a response. Used in EDG Query fields.                                          |
+| [`@edfs__natsSubscribe`](/federation/event-driven-federation/nats)           | Marks a subscription field that listens to one or more NATS subjects. Can include stream configuration.                                            |
+| `@external`                      | Indicates that a field (or all fields of a type) comes from another subgraph and isn’t resolved here.                                              |
+| `@extends`                       | Marks an object or interface as an extension of a type defined in another subgraph, when `extend` isn’t used.                                      |
+| `@inaccessible`                  | Hides a field or type from the API schema while keeping it in the supergraph for internal use.                                                     |
+| `@interfaceObject`               | Lets a subgraph contribute fields to an interface and all implementing types by defining an object abstraction.                                    |
+| `@key`                           | Declares one or more fields that uniquely identify an entity across subgraphs, enabling resolution.                                                |
```

Comment:
has 2 arguments

fields has a type fieldSet - which essentially means a graphql selection set with ommitrd parenthesis {}

resolvable - has default value true

if the resolvable value is false we can't jump to this subgraph to get any fields from it, because it means that given subgraph so not implements entity reference resolver for the given key

There is one exception when we could jump to this subgraph when one of the fields has requires directive we will be able to jump back, but entity reference resolver won't be called, only field resolver